File: base_common/src/main/java/com/lazyxu/base/utils/CommonUtils.java
package com.lazyxu.base.utils;

import android.content.res.Resources;
import android.graphics.Color;
import android.view.View;
import android.view.ViewGroup;
import android.view.ViewParent;

import com.lazyxu.base.base.BaseApplication;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Random;

/**
 * User: xuyexiang
 * Date: 2019/06/13
 * Description:
 * FIXME
 */
class CommonUtils {
    /**
     * 随机颜色
     */
    public static int randomColor() {
        Random random = new Random();
        int red = random.nextInt(150) + 50;//50-199
        int green = random.nextInt(150) + 50;//50-199
        int blue = random.nextInt(150) + 50;//50-199
        return Color.rgb(red, green, blue);
    }

    /**
     * 得到年月日的"日"
     */
    private String getDate() {
        Date date = new Date();
        SimpleDateFormat dateFm = new SimpleDateFormat("dd");
        return dateFm.format(date);
    }


    public static int getColor(int resid) {
        return getResoure().getColor(resid);
    }

    public static Resources getResoure() {
        return BaseApplication.getInstance().getResources();
    }

    public static String[] getStringArray(int resid) {
        return getResoure().getStringArray(resid);
    }

    public static String getString(int resid) {
        return getResoure().getString(resid);
    }

    public static float getDimens(int resId) {
        return getResoure().getDimension(resId);
    }

    public static void removeSelfFromParent(View child) {

        if (child != null) {

            ViewParent parent = child.getParent();

            if (parent instanceof ViewGroup) {

                ViewGroup group = (ViewGroup) parent;

                group.removeView(child);
            }
        }
    }

}

File: base_common/src/main/java/com/lazyxu/base/utils/CommonUtils.kt
package com.lazyxu.base.utils

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.content.res.Resources
import android.text.TextUtils

import com.lazyxu.base.base.BaseApplication

/**
 * User: xuyexiang
 * Date: 2019/06/13
 * Description:
 * FIXME
 */
object CommonUtils {
    /**
    public static Resources getResoure() {
    return BaseApplication.getInstance().getResources();
    }

    public static float getDimens(int resId) {
    return getResoure().getDimension(resId);
    }
     */
    val resoure: Resources
        get() = BaseApplication.getInstance().resources

    fun getDimens(resId: Int): Float {
        return resoure.getDimension(resId)
    }

    fun copy(content: String) {
        if (!TextUtils.isEmpty(content)) {
            val clipboard = BaseApplication.getInstance().getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
            val clip = ClipData.newPlainText(content, content)
            clipboard.primaryClip = clip
        }
    }
}
